src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handlerOperation(FruitTransaction transaction) {
+        int quantityInTransaction = transaction.getQuantity();
+        if (quantityInTransaction <= FruitStorage.getQuantity(transaction.getFruit())) {
+            FruitStorage.fruitTransactionStorage.put(transaction.getFruit(),
+                    FruitStorage.getQuantity(transaction.getFruit())
+                            - quantityInTransaction);
+        } else {
+            throw new RuntimeException("Not enough quantity for sell: fruit - "
+                    + transaction.getFruit()
+                    + " | quantity - "
+                    + transaction.getQuantity());
+        }
```

Comment:
in such cases, throwing/returning early and unnesting the rest of your code is [a preferred approach ](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) (because you don't need `else` after throwing/returning)